x-pack/plugins/security_solution/public/management/pages/event_filters/view/event_filters_list_page.tsx

```diff
@@ -5,20 +5,58 @@
  * 2.0.
  */
 
-import React, { memo, useCallback } from 'react';
+import React, { memo, useCallback, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Dispatch } from 'redux';
+import { useHistory } from 'react-router-dom';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
+import { AppAction } from '../../../../common/store/actions';
+import { getEventFiltersListPath } from '../../../common/routing';
+import { getCurrentLocation, getActionError, getFormEntry } from '../store/selector';
 import { AdministrationListPage } from '../../../components/administration_list_page';
 import { EventFiltersListEmptyState } from './components/empty';
 import { useEventFiltersNavigateCallback, useEventFiltersSelector } from './hooks';
-import { getCurrentLocation } from '../store/selector';
 import { EventFiltersFlyout } from './components/flyout';
 
 export const EventFiltersListPage = memo(() => {
-  const location = useEventFiltersSelector(getCurrentLocation);
+  const history = useHistory();
+  const dispatch = useDispatch<Dispatch<AppAction>>();
+  const isActionError = useEventFiltersSelector(getActionError);
+  const formEntry = useEventFiltersSelector(getFormEntry);
   const navigateCallback = useEventFiltersNavigateCallback();
+  const location = useEventFiltersSelector(getCurrentLocation);
   const showFlyout = !!location.show;
 
+  // Clean url params if wrong
+  useEffect(() => {
+    if ((location.show === 'edit' && !location.id) || (location.show === 'create' && !!location.id))
+      navigateCallback({
+        show: 'create',
+        id: undefined,
+      });
+  }, [location, navigateCallback]);
+
+  // Catch fetch error -> actionError + empty entry in form
+  useEffect(() => {
+    if (isActionError && !formEntry) {
+      // Replace the current URL route so that user does not keep hitting this page via browser back/fwd buttons
+      history.replace(
+        getEventFiltersListPath({
+          ...location,
+          show: undefined,
+          id: undefined,
+        })
+      );
+      dispatch({
+        type: 'eventFiltersFormStateChanged',
+        payload: {
+          type: 'UninitialisedResourceState',
+        },
+      });
+    }
+  }, [dispatch, formEntry, history, isActionError, location, navigateCallback]);
+
```

Comment:
I moved this from flyout in order to have an agnostic url params flyout. This will allow us to rehuse this flyout without depending on the url params